app/routes/user/endorsements.server.ts

```diff
@@ -0,0 +1,352 @@
+   * @param newEndorserEmail - the new user's email for display purposes
+   */
+  async updateRequestedEndorser(
+    existingEndorserSub: string,
+    newEndorserSub: string,
+    newEndorserEmail: string
+  ) {
+    const db = await tables()
+
+    const existingEndorsement = (await db.circular_endorsements.get({
+      requestorSub: this.#sub,
+      endorserSub: existingEndorserSub,
+    })) as EndorsementRequest
+
+    this.#validateStateAndExistanceOfRequest(existingEndorsement)
+
+    await db.circular_endorsements.update({
+      Key: {
+        requestorSub: this.#sub,
+        endorserSub: existingEndorserSub,
+      },
+      UpdateExpression:
+        'set #endorserEmail = :newEndorserEmail, #endorserSub = :newEndorserSub',
+      ExpressionAttributeNames: {
+        '#endorserSub': 'endorserSub',
+        '#endorserEmail': 'endorserEmail',
+      },
+      ExpressionAttributeValues: {
+        ':newEndorserSub': newEndorserSub,
+        ':newEndorserEmail': newEndorserEmail,
+      },
+    })
+  }
+
+  /**
+   * Checks that the current user is in the submitters group and that there is an existing
+   * endorsement request with a key matching the provided requestorSub, the sub of the current
+   * user as the endorserSub, and the status of that request is 'pending'. Then updates the status of that
+   * request to the status parameter provided.
+   * @param status - the new status to be set
+   * @param requestorSub - the sub of the user who requested the endorsement
+   */
+  async updateEndorsementRequestStatus(
+    status: EndorsementState,
+    requestorSub: string
+  ) {
+    if (
+      this.#currentUserGroups.indexOf('gcn.nasa.gov/circular-submitter') == -1
+    )
+      throw new Response(
+        'The user is not a verified submitter, and therefore can not approve requests',
+        {
+          statusText: 'Bad Request',
+          status: 400,
+        }
+      )
+
+    const db = await tables()
+
+    const existingEndorsement = (await db.circular_endorsements.get({
+      requestorSub: requestorSub,
+      endorserSub: this.#sub,
+    })) as EndorsementRequest
+
+    this.#validateStateAndExistanceOfRequest(existingEndorsement)
+
+    await db.circular_endorsements.update({
+      Key: {
+        requestorSub: requestorSub,
+        endorserSub: this.#sub,
+      },
+      UpdateExpression: 'set #status = :status',
+      ExpressionAttributeNames: {
+        '#status': 'status',
+      },
+      ExpressionAttributeValues: {
+        ':status': status,
+      },
+    })
+  }
+
+  /**
+   * Allows the current user to delete the existing endorsement request corresponding to
+   * the provided endorserSub.
+   * @param endorserSub - the sub of the listed endorser on the existing request
+   */
+  async deleteEndorsementRequest(endorserSub: string) {
+    const db = await tables()
+    await db.circular_endorsements.delete({
+      requestorSub: this.#sub,
+      endorserSub,
+    })
+  }
```

Comment:
Only permit deleting "pending" requests.

Reply:
Check with a ConditionExpression

Reply:
Do you have docs or examples of what you are talking about using?

Reply:
`(await tables())._db` ...